cmd/cluster/health.go

```diff
@@ -3,74 +3,55 @@ package cluster
 import (
 	"fmt"
 	"log"
-	"os"
 	"strconv"
 	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
-	"github.com/openshift/osdctl/internal/utils/globalflags"
-	k8spkg "github.com/openshift/osdctl/pkg/k8s"
-	awsprovider "github.com/openshift/osdctl/pkg/provider/aws"
+	"github.com/openshift/osdctl/pkg/osdCloud"
 	"github.com/openshift/osdctl/pkg/utils"
 	"github.com/spf13/cobra"
 	"gopkg.in/yaml.v2"
-	"k8s.io/cli-runtime/pkg/genericclioptions"
-	"k8s.io/klog/v2"
 	cmdutil "k8s.io/kubectl/pkg/cmd/util"
 )
 
 // healthOptions defines the struct for running health command
 // This command requires the ocm API Token https://cloud.redhat.com/openshift/token be available in the OCM_TOKEN env variable.
 
 type healthOptions struct {
-	k8sclusterresourcefactory k8spkg.ClusterResourceFactoryOptions
-	output                    string
-	verbose                   bool
-
-	genericclioptions.IOStreams
-	GlobalOptions *globalflags.GlobalOptions
+	clusterID  string
+	output     string
+	verbose    bool
+	awsProfile string
 }
 
 // newCmdHealth implements the health command to describe number of running instances in cluster and the expected number of nodes
-func newCmdHealth(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
-	ops := newHealthOptions(streams, flags, globalOpts)
+func newCmdHealth() *cobra.Command {
+	ops := newHealthOptions()
 	healthCmd := &cobra.Command{
 		Use:               "health",
-		Short:             "Describes health of cluster nodes and provides other cluster vitals. Requires being logged into the cluster's corresponding hive cluster.",
+		Short:             "Describes health of cluster nodes and provides other cluster vitals.",
 		Args:              cobra.NoArgs,
 		DisableAutoGenTag: true,
 		Run: func(cmd *cobra.Command, args []string) {
 			cmdutil.CheckErr(ops.complete(cmd, args))
 			cmdutil.CheckErr(ops.run())
 		},
 	}
-	ops.k8sclusterresourcefactory.AttachCobraCliFlags(healthCmd)
+
 	healthCmd.Flags().BoolVarP(&ops.verbose, "verbose", "", false, "Verbose output")
+	healthCmd.Flags().StringVarP(&ops.clusterID, "cluster-id", "C", "", "Cluster ID")
+	healthCmd.Flags().StringVarP(&ops.awsProfile, "profile", "p", "", "AWS Profile")
 	healthCmd.MarkFlagRequired("cluster-id")
 	return healthCmd
 }
 
-func newHealthOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *healthOptions {
-	return &healthOptions{
-		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
-			Flags: flags,
-		},
-		IOStreams:     streams,
-		GlobalOptions: globalOpts,
-	}
+func newHealthOptions() *healthOptions {
```

Comment:
minor: Can probably remove this func now

Reply:
I'm leaving it for now so it matches all our commands. I'm planning on cleaning these up across all commands as part of the larger refactor eventually.